doc/src/build/programming-with-objects/ch5-child-objects.md

```diff
@@ -55,57 +53,59 @@ Similarly, we can define an API to create an object of `Parent` type:
-If we comment out the second line, the Move compiler will complain that we cannot drop `child_ref`.
+This function takes `child` by value, fills the `child` field of `parent` with the `ID` of the `Child` object, and calls `transfer_to_object` to transfer the `child` object to the `parent`.
 At the end of the `add_child` call, we have the following ownership relationship:
-1. Sender account address owns a `Parent` object.
+1. Sender account address (still) owns a `Parent` object.
 2. The `Parent` object owns a `Child` object.
 
 #### transfer_to_object_id
-In the above example, `Parent` has an optional child field. What if the field is not optional? We must construct `Parent` with a `ChildRef`. However, in order to have a `ChildRef`, we have to transfer the child object to the parent object first. This creates a paradox. We cannot create the parent unless we have a `ChildRef`, and we cannot have a `ChildRef` unless we already have the parent object. To solve this exact problem and be able to construct a non-optional `ChildRef` field, we provide another API that allows you to transfer an object to object ID, instead of to object:
+In the above example, `Parent` has an optional child field. What if the field is not optional? We must construct `Parent` with a `ID`. However, in order to have a valid `ID`, we have to transfer the child object to the parent object first. This creates a somewhat paradoxical situation. We cannot create the parent unless we have a valid `ID`, and we cannot have a valid `ID` unless we already have the parent object. To solve this problem, we can use a different API that allows you to transfer an object to an object ID, instead of to the object itself:
 ```rust
 public fun transfer_to_object_id<T: key>(
     obj: T,
-    owner_id: VersionedID,
-): (VersionedID, ChildRef<T>);
+    owner_id: &mut UID,
+);
 ```
-To use this API, we don't need to create a parent object yet; we need only the object ID of the parent object, which can be created in advance through `object::new(ctx)`. The function returns a tuple: it will return the `owner_id` that was passed in, along with the `ChildRef` representing a reference to the child object `obj`. It may seem strange that we require passing in `owner_id` by value only to return it. This is to ensure that the caller of the function does indeed own a `Info` that hasn't been used in any object yet. Without this, it can be easy to make mistakes.
+To use this API, we don't need to create a parent object yet; we need only the `UID` of the parent object, which can be created in advance through `object::new(ctx)`. The function requres a mutable reference to the parent `UID` for two reasons. (1) it prevents children from being added to immutable objects (more on that later). (2) it gives the module that defines the parent object more control. Namely, it can expose a function to get an immutable reference to the `&UID` without worrying about external caller adding child objects.
```

Comment:
```suggestion
To use this API, we don't need to create a parent object yet; we need only the `UID` of the parent object, which can be created in advance through `object::new(ctx)`. The function requires a mutable reference to the parent `UID` for two reasons. (1) it prevents children from being added to immutable objects (more on that later). (2) it gives the module that defines the parent object more control. Namely, it can expose a function to get an immutable reference to the `&UID` without worrying about external caller adding child objects.
```